src/btree/bt_cursor.c

```diff
@@ -308,12 +305,17 @@ __wt_cursor_valid(WT_CURSOR_BTREE *cbt, WT_ITEM *key, uint64_t recno, bool *vali
     if (cbt->ins != NULL) {
         if (WT_CURSOR_BOUNDS_SET(&cbt->iface)) {
             /* Get the insert list key. */
-            if (key == NULL && btree->type == BTREE_ROW) {
-                tmp_key.data = WT_INSERT_KEY(cbt->ins);
-                tmp_key.size = WT_INSERT_KEY_SIZE(cbt->ins);
-            }
-            WT_RET(__btcur_bounds_contains_key(
-              session, &cbt->iface, key != NULL ? key : &tmp_key, &key_out_of_bounds, NULL));
+            if (btree->type == BTREE_ROW) {
+                if (key == NULL) {
+                    tmp_key.data = WT_INSERT_KEY(cbt->ins);
+                    tmp_key.size = WT_INSERT_KEY_SIZE(cbt->ins);
+                }
```

Comment:
We could remove this ternary if by calling this once within the key == NULL if statement and once in the else condition. Would that be better?

Reply:
Hmm not quite sure if it is cleaner 
```
                /* Get the insert list key. */
                if (key == NULL) {
                    tmp_key.data = WT_INSERT_KEY(cbt->ins);
                    tmp_key.size = WT_INSERT_KEY_SIZE(cbt->ins);
                    WT_RET(__btcur_bounds_contains_key(session, &cbt->iface, &tmp_key, WT_RECNO_OOB, &key_out_of_bounds, NULL));
                } else 
                    WT_RET(__btcur_bounds_contains_key(session, &cbt->iface, key, WT_RECNO_OOB, &key_out_of_bounds, NULL));
```
what do you think @luke-pearson ?

Reply:
I say go with this until we have a better solution to passing the key and the recno everywhere, at least now we have one less ternary if.